allauth/account/models.py

```diff
@@ -48,7 +48,7 @@ def set_as_primary(self, conditional=False):
         self.primary = True
         self.save()
         user_email(self.user, self.email)
-        self.user.save()
+        self.user.save(update_fields=[app_settings.USER_MODEL_EMAIL_FIELD])
```

Comment:
**Issue**: `USER_MODEL_EMAIL_FIELD` can be `None` in case the model has no such field.

Reply:
Thanks @pennersr !
But overall PR makes sense or you don't see sense in minor changes like that?

Reply:
Great, I will update this PR then.

Reply:
I wrote tests and fix condition when `USER_MODEL_EMAIL_FIELD` is `None`